src/ParticleDA.jl

```diff
@@ -3,19 +3,22 @@ module ParticleDA
 using Distributions, Statistics, MPI, Base.Threads, YAML, HDF5
 using TimerOutputs
 
-export run_particle_filter, BootstrapFilter
+export run_particle_filter, BootstrapFilter, OptimalFilter
 
 include("params.jl")
 include("io.jl")
+include("OptimalFilter.jl")
 
 using .Default_params
 
 # Functions to extend in the model
 
 """
-    ParticleDA.get_grid_size(model_data) -> NTuple{N, Int} where N
+    ParticleDA.get_grid_size(model_data) ->
+        NamedTuple({:nx, :ny, :dx, :dy, :x_length, :y_length},Tuple{Int, Int, Float, Float, Float, Float})
 
-Return a tuple with the size of the data.
+Return a NamedTuple with the parameters that define the size of the grid.
+TODO: It could be better to define a struct for these
```

Comment:
I think functions with different names might be a cleaner solution here. Ie, function `f(d) -> NTuple{N,Int}` that returns (nx,ny,...) and `g(d) -> NTuple{N,Float}` that returns (x_length, y_length, ...). dx and dy can be derived from these so a third function is not really needed (although it might help avoid errors). I'm trying to choose names would be most descriptive. Does this sound right to you:

- `get_grid_dims`: get nx,ny,...
- `get_grid_size`: get x_length, y_length,...
- `get_grid_cell_size`: get dx, dy,...

Reply:
See #157